### PR TITLE
BackEnd apis integration with mongoDB and minor bug fixes in the codebases

### PR DESCRIPTION
**Pull request by jeeva1429**
Feat: Integrated the backend apis with mongoDB. 
> Fix: Fix some of the bugs in the hackathon creation form

### CODE REVIEW

File: backend/server/server.js [hunk 1]
```diff
@@ -1,29 +1,92 @@
...
+const multer = require("multer");
+const upload = multer({ storage: multer.memoryStorage() });
+const axios = require("axios");
+const FormData = require("form-data");
+const mockedData = require("../../frontEnd/src/components/data.json");
+const mongoose = require("mongoose");
+const Hackathon = require("./HackathonModel");
+const apiKey = process.env.API_KEY;
+app.use(cors());
+console.log(process.env.MONGODB_STR);
```
Comment (yash0212) [inline backend/server/server.js:15]: Console log can be removed, this can expose your mongodb connection string on server

File: frontEnd/src/assets/Screenshot 2023-05-25 211230.png
Comment (yash0212) [inline frontEnd/src/assets/Screenshot 2023-05-25 211230.png:1]: Please rename the file to the standardized format

File: frontEnd/src/components/hackathonDetails/details.jsx [hunk 2]
```diff
@@ -7,53 +7,60 @@ import HandleDay from '../hackathonLister/handleDay';
 import HandleTime from '../hackathonLister/handleTime';
 
 function Details() {
-    const { id } = useParams();
+    const { id  } = useParams();
     const [hackathon,setHackathon]=useState([])
 
+    console.log(hackathon.Themes?.[0]);
```
Comment (yash0212) [inline frontEnd/src/components/hackathonDetails/details.jsx:13]: Console log can be removed

File: frontEnd/src/components/hackathonDetails/details.jsx [hunk 3]
```diff
@@ -7,53 +7,60 @@ import HandleDay from '../hackathonLister/handleDay';
...
                 </div>
                 <div className="flex gap-2 items-center text-emperor font-bold text-sm mt-5">
                     <span>
-                        <HandleDate date={hackathon.date} format={'short'} />
+                        <HandleDate date={hackathon.Date} format={'short'} />
                     </span>
                     <img src={fullStop} className="h-4" />
-                    {hackathon.tags?.map((data) => <span key={data}>{data}</span>)}
-                     
+                    {hackathon.Themes && hackathon?.Themes[0].split(',').map((tag) => (
```
Comment (yash0212) [inline frontEnd/src/components/hackathonDetails/details.jsx:42]: Is the split by `,` required? I guess there is a logical error here which takes only the 1st entry from the `Themes` array

File: frontEnd/src/components/hackathonLister/hackathonCardSkeletol.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';
+
+function HackathonCardSkeleton() {
+    return (
+        <>
```
Comment (yash0212) [inline frontEnd/src/components/hackathonLister/hackathonCardSkeletol.jsx:6]: React Fragment(`<> </>`) can be removed as it serves no purpose here

File: package-lock.json
Comment (yash0212) [inline package-lock.json:1]: Please explain the use of this file

File: package.json
Comment (yash0212) [inline package.json:1]: Please explain the use of this file

**Review by yash0212 — CHANGES_REQUESTED** [review]

File: package.json
Comment (jeeva1429) [reply package.json:1]: It contains important information about the project, such as its name, version, and description. It also lists all the external packages (dependencies) that the project needs to work correctly.

File: frontEnd/src/components/hackathonDetails/details.jsx (hunk 3 above)
Comment (jeeva1429) [reply frontEnd/src/components/hackathonDetails/details.jsx:42]: The value of Themes is an array of themes like ['ai','ml']. To split that , I used split(',')

File: package-lock.json
Comment (jeeva1429) [reply package-lock.json:1]: The package-lock.json file is automatically generated and maintained by the npm package manager when installing or updating packages in a Node.js project. It serves as a lockfile, which means it keeps a record of the exact versions of all the packages installed in the project.

File: package.json
Comment (yash0212) [reply package.json:1]: Ok, so why does it exist out of `backEnd` or `frontEnd` folder

File: package-lock.json
Comment (yash0212) [reply package-lock.json:1]: Ok, so why does it exist out of `backEnd` or `frontEnd` folder?

File: frontEnd/src/components/hackathonDetails/details.jsx (hunk 3 above)
Comment (yash0212) [reply frontEnd/src/components/hackathonDetails/details.jsx:41]: If the value of `Themes` is an array then split wouldn't work as split works on a `String` object. Ref - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split

File: backend/models/dbConnection.js
Comment (yash0212) [inline backend/models/dbConnection.js:1]: `models` is not the ideal directory to store the `dbConnection.js` file. Either have a `config` folder inside `backend` or keep the `dbConnection.js` as the direct child of `backend` folder

File: frontEnd/src/App.jsx [hunk 5]
```diff
@@ -3,26 +3,16 @@ import { Routes, Route } from 'react-router';
 import Homepage from './components/pages/homepage';
 import Details from './components/hackathonDetails/details';
 import HackathonCreation from './components/hackathonCreator/hackathonCreation';
+// import HackathonForm from './components/hackathonCreator/hackathonForm';
```
Comment (yash0212) [inline frontEnd/src/App.jsx:6]: Commented code can be removed

File: frontEnd/src/components/filters/filters.jsx [hunk 6]
```diff
@@ -26,13 +27,13 @@ function Filters({hackathon}) {
...
      };
-     useEffect(() => {
+      useEffect(() => {
          let filteredHackathons = hackathon;
 
          // Filter by theme
          themeFilter !== ''
              ? (filteredHackathons = filteredHackathons.filter((hackathon) =>
-                   hackathon.tags.includes(themeFilter.toLowerCase())
+                   hackathon.Themes[0].includes(themeFilter.toLowerCase())
```
Comment (yash0212) [inline frontEnd/src/components/filters/filters.jsx:36]: Having `Themes[0]` does not follow your data model. Please check for logical errors while saving data in `Themes`. It seems like your logic is saving all the themes as comma separated values at `0th` index of the `Themes` array

**Review by yash0212 — CHANGES_REQUESTED** [review]

File: frontEnd/src/App.jsx (hunk 5 above)
Comment (yash0212) [reply frontEnd/src/App.jsx:6]: Please resolve this

File: frontEnd/src/components/filters/filters.jsx (hunk 6 above)
Comment (yash0212) [reply frontEnd/src/components/filters/filters.jsx:36]: Please update this according to new changes

File: package-lock.json
Comment (yash0212) [inline package-lock.json:1]: Is this file required?

File: package.json
Comment (yash0212) [inline package.json:1]: Is this file required?

**Review by yash0212 — CHANGES_REQUESTED** [review]
Please resolve the above comments

**Review by yash0212 — APPROVED** [review]
